### PR TITLE
Webcomponents on published Projects

### PR DESCRIPTION
**Pull request by mjschranz**

### CODE REVIEW

**Comment by mjschranz** [comment]
Ignore the majority of this pull request for now unless you feel like playing with the demo pages.

**Comment by Pomax** [comment]
note: this pull request should not be merged. It's too experimental, and exist solely for review.

**Comment by Pomax** [comment]
It looks like you've included the actual webcomponent files, rather than submodule-ing the toolkitchen repos for web components and custom elements. It's better to submodule to those instead.

**Comment by mjschranz** [comment]
And I shall be eventually. That's in the next step once I finish writing the rest of the web components for our plugins.

File: public/templates/web-components/webmaker-components.html [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+    // our element template:
+    var template = this.querySelector( "template" );
+
+    // our "what to do when this element is built" code:
+    this.register({
+      prototype: {
+        readyCallback: function() {
+          var text = this.innerHTML,
+              defaultDecorations = { bold: false, italics: false },
+              start = this.attributes.getNamedItem( "start" ) ? this.attributes.getNamedItem( "start" ).nodeValue : 0,
```
Comment (Pomax) [inline public/templates/web-components/webmaker-components.html]: inside readycallback:

```
var options = {},
attributeNames = element.getAttribute("attributes");
attributeNames.forEach(function(attr) { 
  options[attr] = this.attributes.getNamedItem(attr);
});
```

would probably make more sense

**Comment by mjschranz** [comment]
Closing in favor of a new branch.

**Pull request closed without merging**